main.go

```diff
@@ -0,0 +1,209 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/fatih/structs"
+	"github.com/iancoleman/strcase"
+	"github.com/sirupsen/logrus"
+	"github.com/vshn/appcat-cmd/internal/defaults"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/printers"
+)
+
+func init() {
+	logrus.SetOutput(os.Stderr)
+}
+
+func main() {
+	code := Main(os.Args, os.Stdin, os.Stdout)
+	os.Exit(code)
+}
+
+func Main(args []string, in io.Reader, out io.Writer) int {
+	plainArgs := args
+	service, err := findServiceType(plainArgs[1])
+	if err != nil {
+		logrus.Error(err)
+		return 1
+	}
+
+	plainArgs, err = cleanInputArguments(plainArgs)
+	if err != nil {
+		logrus.Error(err)
+		return 1
+	}
+
+	parameters := parseArgs(plainArgs)
+
+	_, err = decorateType(service, parameters)
+	if err != nil {
+		logrus.Error(err)
+		return 1
+	}
+
+	err = writeYAML(service, os.Stdout)
+	if err != nil {
+		logrus.Error(err)
+		return 1
+	}
+
+	return 0
+}
+
+func writeYAML(service runtime.Object, out io.Writer) error {
+	yp := printers.YAMLPrinter{}
+	err := yp.PrintObj(service, out)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Split input parameter of type "--foo=bar" to ["--foo", "bar"]
+func splitArguments(argument string) ([2]string, error) {
+	if strings.Count(argument, "=") > 1 {
+		err := fmt.Errorf("argument %s is invalid", argument)
+		return [2]string{}, err
+	}
+	splitArgs := strings.Split(argument, "=")
+	return [2]string(splitArgs), nil
+}
+
+// Checks and fixes if input parameters are of type "--foo=bar"
+// returns fixed argument list
+func cleanInputArguments(arguments []string) ([]string, error) {
+	var fixedArguments []string
+	copy(fixedArguments, arguments)
+	for _, argument := range arguments {
+		if strings.Contains(argument, "=") {
+			fixed, err := splitArguments(argument)
+			if err != nil {
+				return nil, err
+
+			}
+			fixedArguments = append(fixedArguments, fixed[0], fixed[1])
+		} else {
+			fixedArguments = append(fixedArguments, argument)
+		}
+	}
+	return fixedArguments, nil
+}
+
+// Takes the input arguments and outputs them as separate values
+// parameterName = "Parent.child1.child2. ... .childn"
+// [parameterName-1 value-1, ...., parameterName-n value-n ] -> [[parent-1, child-1, ...child-n, value1],...]
+func parseArgs(args []string) [][]string {
```

Comment:
I feel that this implements stuff that the "github.com/spf13/pflag" library would do for us. Is there any specific reason why we're not using that library?

Reply:
Yes, since the flag itself represents the hierarchical "path" of the value to be set(I couldn't find how to make use of that with pflags), we don't have an fixed amount of valid input flags and we don't actually have to control behavior of appcat-cli through flags.

Reply:
Makes a lot of sense, thanks!